sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -263,4 +281,108 @@ case class InsertIntoHadoopFsRelationCommand(
       }
     }.toMap
   }
+
+  /**
+   * Detect the conflict when there are several InsertIntoHadoopFsRelation operations
+   * write concurrently.
+   */
+  private def detectConflict(
+      fs: FileSystem,
+      path: Path,
+      isOverwrite: Boolean,
+      staticPartitionKVs: Seq[(String, String)]): Unit = {
+    for (i <- 0 until partitionColumns.size) {
+      val checkedStagingPath = ListBuffer(s".spark-staging-overwrite-$i")
+      if (isOverwrite) {
+        checkedStagingPath += s".spark-staging-append-$i"
```

Comment:
what about two concurrent job both appending data to the same partitions? 

In current approach, we didn't detect the conflict? I think we have to disable concurrent append to the same partition also.

Reply:
Sounds good.
So we can remove the overwrite flag for staging dir. 
And we no longer need  `isOverwrite`  parameter.